### PR TITLE
Allow to list tags in hawkular-metrics

### PR DESCRIPTION
**Pull request by rubenvp8510**
Need to add specs.

### CODE REVIEW

File: lib/hawkular/metrics/metric_api.rb [hunk 1]
```diff
@@ -69,6 +69,18 @@ def query_stats(gauge_ids: [], counter_ids: [], avail_ids: [], rates: false, sta
       http_post(path, data)
     end
 
+    # Fetch all tags for metrics definitions
+    # @return [Hash{String=>String}]
+    def tags
+      tags = []
+      http_get('/metrics/').map do |g|
+        g['tags'].map do |k, v|
```
Comment (pilhuhn) [inline lib/hawkular/metrics/metric_api.rb:77]: Can you please check with e.g. @burmanm about tags with a list of values ?

File: lib/hawkular/metrics/metric_api.rb [hunk 2]
```diff
@@ -69,6 +69,18 @@ def query_stats(gauge_ids: [], counter_ids: [], avail_ids: [], rates: false, sta
       http_post(path, data)
     end
 
+    # Fetch all tags for metrics definitions
+    # @return [Hash{String=>String}]
+    def tags
```
Comment (pilhuhn) [inline lib/hawkular/metrics/metric_api.rb:74]: Would it make sense to allow limiting by key - e.g. 'only return tags that start with app:' ?

File: lib/hawkular/metrics/metric_api.rb (hunk 1 above)
Comment (burmanm) [reply lib/hawkular/metrics/metric_api.rb:77]: We don't currently support single tag with multiple values. It's currently a solution dependant on how to do this, for example in Openshift integration we do with comma separated list. If there's a need for such in this solution, could you create a JIRA ticket about it? We could then look at implementing it in some sane way..

**Comment by pilhuhn** [comment]
On 17 Oct 2016, at 9:05, Michael Burman wrote:

> We don't currently support single tag with multiple values.

This is ok then. I just saw the openshift thing.
Perhaps we should support this here as well (?)

**Comment by pilhuhn** [comment]
Thanks @rubenvp8510

**Pull request merged**